test/LiquidityManager/invest.test.ts

```diff
@@ -117,12 +115,97 @@ describe.only('LiquidityManager#invest', () => {
 
         const amountBn = new BigNumber(amount.toString())
 
+        // Min output considering that we need to swap the token before
         return Slippage.deductSlippage(
             BigInt(amountBn.div(tokenPrice).toFixed(0)),
             SLIPPAGE_BN.times(2),
         )
     }
 
+    async function investAndValidateTransaction(
+        pool: Pool,
+        amount0: bigint,
+        amount1: bigint,
+        token0: ERC20Priced,
+        token1: ERC20Priced,
+        swapAmountToken0: bigint,
+        swapAmountToken1: bigint,
+        tickLower: number,
+        tickUpper: number,
+    ) {
+        const [swapToken0, swapToken1] = await Promise.all([
+            getEncodedSwap(swapAmountToken0, token0),
+            getEncodedSwap(swapAmountToken1, token1),
+        ])
+
+        const receipt = await (await liquidityManager
+            .connect(account0)
+            .investUniswapV3(
+                {
+                    positionManager: positionManagerUniV3,
+                    inputToken: stablecoin,
+                    depositAmountInputToken: amount,
+
+                    fee: pool.fee,
+
+                    token0: pool.token0.address,
+                    token1: pool.token1.address,
+
+                    swapToken0,
+                    swapToken1,
+
+                    swapAmountToken0,
+                    swapAmountToken1,
+
+                    tickLower,
+                    tickUpper,
+
+                    amount0Min: getMinOutput(swapAmountToken0, token0.price),
+                    amount1Min: getMinOutput(swapAmountToken1, token1.price),
+                },
+                permitAccount0,
+            )).wait()
+
+        validateInvestTransaction(receipt, amount0, token0.price, amount1, token1.price)
```

Comment:
why make this a separate function if only called here?

Reply:
Separation of responsibilities i guess... :rofl:  
Maybe i could call it outside of the invest function or merge both methods